FileNotFoundException because device name contains a colon

Hi, I've got a FileNotFoundException on the file `com.squareup.spoon.html.HtmlRenderer` and at this method `renderMustacheToFile`.  [Link](https://github.com/square/spoon/blob/master/spoon-runner/src/main/java/com/squareup/spoon/html/HtmlRenderer.java#L152)

I think the problem comes from the fact I'm using a remote device (by Genymotion) and the URL of the device (the name) contains a colon `:`

Is there a way to rename it or to avoid it ?

Thanks a lot for this great tool by the way ;)

Stacktrace :

```
2013-07-08 15:10:35 [SR.runTests] Executing instrumentation suite on 1 device(s).
Exception in thread "main" java.lang.RuntimeException: java.io.FileNotFoundException: spoon-output\logs\192.168.56.101:5555\com.example.simple.test.MainActivityTest\testBlankPassword_ShowsError.html (
La syntaxe du nom de fichier, de rÚpertoire ou de volume est incorrecte)
        at com.squareup.spoon.html.HtmlRenderer.renderMustacheToFile(HtmlRenderer.java:150)
        at com.squareup.spoon.html.HtmlRenderer.generateLogHtml(HtmlRenderer.java:138)
        at com.squareup.spoon.html.HtmlRenderer.render(HtmlRenderer.java:58)
        at com.squareup.spoon.SpoonRunner.run(SpoonRunner.java:80)
        at com.squareup.spoon.SpoonRunner.main(SpoonRunner.java:376)
Caused by: java.io.FileNotFoundException: spoon-output\logs\192.168.56.101:5555\com.example.simple.test.MainActivityTest\testBlankPassword_ShowsError.html (La syntaxe du nom de fichier, de rÚpertoire
ou de volume est incorrecte)
        at java.io.FileOutputStream.open(Native Method)
        at java.io.FileOutputStream.<init>(Unknown Source)
        at java.io.FileOutputStream.<init>(Unknown Source)
        at java.io.FileWriter.<init>(Unknown Source)
        at com.squareup.spoon.html.HtmlRenderer.renderMustacheToFile(HtmlRenderer.java:147)
        ... 4 more
```

--- a/spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java
+++ b/spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java
@@ -52,8 +52,14 @@ public final class SpoonRunner {
     this.debug = debug;
     this.className = className;
     this.methodName = methodName;
-    this.serials = unmodifiableSet(serials);
     this.classpath = classpath;
+
+    // Sanitize the serials for use on the filesystem as a folder name.
+    Set<String> serialsCopy = new LinkedHashSet<String>(serials.size());
+    for (String serial : serials) {
+      serialsCopy.add(SpoonUtils.sanitizeSerial(serial));
+    }
+    this.serials = unmodifiableSet(serialsCopy);
   }
 
   /**

--- a/spoon-runner/src/main/java/com/squareup/spoon/SpoonUtils.java
+++ b/spoon-runner/src/main/java/com/squareup/spoon/SpoonUtils.java
@@ -19,6 +19,7 @@ import java.lang.reflect.InvocationTargetException;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.regex.Pattern;
 import javax.imageio.ImageIO;
 import org.apache.commons.io.FileUtils;
 
@@ -28,6 +29,7 @@ import static com.android.ddmlib.Log.LogLevel.DEBUG;
 
 /** Utilities for executing instrumentation tests on devices. */
 final class SpoonUtils {
+  private static final Pattern SERIAL_VALIDATION = Pattern.compile("[^a-zA-Z0-9_-]");
   static final Gson GSON = new GsonBuilder() //
       .registerTypeAdapter(File.class, new TypeAdapter<File>() {
         @Override public void write(JsonWriter jsonWriter, File file) throws IOException {
@@ -55,6 +57,10 @@ final class SpoonUtils {
       }
     }
     throw new IllegalArgumentException("Unknown device serial: " + serial);
+  }
+
+  static String sanitizeSerial(String serial) {
+    return SERIAL_VALIDATION.matcher(serial).replaceAll("_");
   }
 
   /** Get a {@link FileEntry} for an arbitrary path. */

--- a/spoon-runner/src/test/java/com/squareup/spoon/SpoonUtilsTest.java
+++ b/spoon-runner/src/test/java/com/squareup/spoon/SpoonUtilsTest.java
@@ -1,0 +1,14 @@
+package com.squareup.spoon;
+
+import org.junit.Test;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+public class SpoonUtilsTest {
+  @Test public void serialSanitizer() {
+    assertThat(SpoonUtils.sanitizeSerial("1234")).isEqualTo("1234");
+    assertThat(SpoonUtils.sanitizeSerial("FooBarBaz")).isEqualTo("FooBarBaz");
+    assertThat(SpoonUtils.sanitizeSerial("ST-398H984")).isEqualTo("ST-398H984");
+    assertThat(SpoonUtils.sanitizeSerial("10.0.0.1:1234")).isEqualTo("10_0_0_1_1234");
+  }
+}

